src/main/java/com/passpnu/passwordmanager/config/WebSecurityConfiguration.java

```diff
@@ -0,0 +1,37 @@
+package com.passpnu.passwordmanager.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.authentication.AuthenticationManager;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfiguration{
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception{
+        http.csrf().disable().authorizeHttpRequests(
+                (requests) -> requests.requestMatchers("/log-in", "/sign-up", "/passwords/generate").permitAll()
```

Comment:
Should password generator be accessible for all users or just the authorized ones? Because it seems to me that only login and sign-up may be not protected

Reply:
This method only generates a random string and nothing all. Every guest have access for this method, it's also described in my system operations for the project